lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -6,6 +6,32 @@ class JsonApi < Base
         autoload :PaginationLinks
         autoload :FragmentCache
 
+        module ApiObjects
+          # Make JSON API top-level jsonapi member opt-in
+          # ref: http://jsonapi.org/format/#document-top-level
+          ActiveModel::Serializer.config.jsonapi_include_toplevel_member = false
+          ActiveModel::Serializer.config.jsonapi_version = '1.0'
+          ActiveModel::Serializer.config.jsonapi_toplevel_meta = {}
+          module JsonApi
+            module_function
+
+            def include_member?
+              ActiveModel::Serializer.config.jsonapi_include_toplevel_member
+            end
+
+            def add!(document)
```

Comment:
With the former, in case of bug, you have to check how exactly is `do_some_magic` modifying its argument, whereas with the latter, you can directly see whether it potentially has anything to do with it at all.